### PR TITLE
Add destination field to AuthnRequest

### PR DESCRIPTION
**Pull request by Philonous**
**Summary**
`AuthnRequest` has an optional attribute, `Destination`, which is however required for signed requests. This PR adds this attribute.

**Checklist**

- [x] All definitions are documented with Haddock-style comments.
- [x] All exported definitions have `@since` annotations.
- [x] Code is formatted in line with the existing code.
- [x] The changelog has been updated.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog for `wai-saml2`
 
+## 0.4.1
+
+- Added `authnRequestDestination` field to `AuthnRequest`
```
Comment (mbg) [inline CHANGELOG.md]: I think this probably warrants a major version change to comply with the PvP, but for now it is OK to just use "Unreleased changes" as the header here:

```suggestion
## Unreleased changes

- Added `authnRequestDestination` field to `AuthnRequest`
```

**Review by mbg — APPROVED** [review]
Looks good to me! Thank you for adding this ✨

There's just comment about the versioning. Let me know if you want to act on that - otherwise, I am happy to merge this as is and then deal with the versioning issue myself later.

File: CHANGELOG.md (hunk 1 above)
Comment (mbg) [reply CHANGELOG.md]: I have made this change now and resolved the merge conflict.

**Comment by Philonous** [comment]
Ah, thanks for taking care of this!

**Pull request merged**